### ISSUE TITLE
[Boled] Cross FB Layer indication : 'Effects From' no has no impact

### ISSUE THREAD

**Issue by majjam** [nonlinear-labs-dev/C15#2136]
(Please close issue  if Master is not intended to reflect the Release-branch)
On layered sound in **FB Mixer** at parameter **Effects - Fom I** with values != 0 there is no cross-FB-indication in sound overview screen.

Software version:
![grafik](https://user-images.githubusercontent.com/4758563/90312051-98d0bf00-df01-11ea-8f28-28f80048989e.png)

Setting:
![grafik](https://user-images.githubusercontent.com/4758563/90312036-7474e280-df01-11ea-85a2-c9e779ad4290.png)

Sound screen result:
![grafik](https://user-images.githubusercontent.com/4758563/90312089-f6650b80-df01-11ea-9812-55cfd3f11cd7.png)

I understand indication should be as following : 
![grafik](https://user-images.githubusercontent.com/4758563/90312168-958a0300-df02-11ea-9ce6-114f9f0d0cfb.png)

**Comment by MatthiasSeeber** [comment]
@majjam

As far as I can tell, the observed behavior is _intentional_. Consider a blank Layer Sound (every FB Mixer Channel is 0.0%, as well as any **From** Parameter). The resulting Sound Screen should look like this:

![Boled-Mon-Aug-17-12:22:25-2020
](https://user-images.githubusercontent.com/37212609/90386387-c8f49b00-e084-11ea-87fb-f23d8613213f.png)

Notice, that no FB arrows are shown - as there isn't any. Changing the **Effects - From II** now

![Boled-Mon-Aug-17-12:22:50-2020
](https://user-images.githubusercontent.com/37212609/90386458-e590d300-e084-11ea-9e6f-f50e5f2007d3.png)

will not change any display of arrows, because the Channel Level is also evaluated here. As long as the level is 0.0%, the **From** parameter on its own has no effect. If now, the **Effects - Level** is set unequal to 0.0%

![Boled-Mon-Aug-17-12:23:11-2020
](https://user-images.githubusercontent.com/37212609/90386660-2daff580-e085-11ea-8d22-88cc2ae49c5d.png)

this will produce two arrows, as Self and Cross FB will now be able to flow.

![Boled-Mon-Aug-17-12:23:17-2020
](https://user-images.githubusercontent.com/37212609/90386716-428c8900-e085-11ea-8893-ad44ac990b6a.png)

So, in short: Self FB arises from Channel Levels unequal to 0.0%, Cross FB arises from the corrsponding **From** as well as Channel Levels.

If you can verify this, I suggest to close this issue.

**Comment by majjam** [comment]
@MatthiasSeeber 
thanks for clarification. My test case reacts exactly like you described
My first thought was that 'Effects From I/II' included the 'Rev mix', too and not only the other effects.
So am I right that there is no cross FB for the reverb ?

If that is the case this issue can be closed safely.

**Comment by MatthiasSeeber** [comment]
Effects and Reverb Mix both depend on the one corresponding From parameter. It is helpful to regard the Effects as a Mixer Channel and the Reverb Mix as a part of that channel.

Both Part Effect Groups produce a pre and post reverb signal. The Effects From parameter now serves for crossfading both pre I / II and post I / II. Those two resulting signals serve for the Reverb Mix crossfade. Regarding the cross feedback signal flow, the Reverb Mix doesn't affect any arrows, it just blends the already crossfaded Part effects between pre and post reverb.

So, for a Layer Sound, self feedback can arise from the channels Comb, SVF and Effects - condition is that the channel level is unqeual 0.0% and that the corresponding From parameter is less than 100.0%. Cross feedback can arise from the channels A/B, Comb, SVF and Effects - condition is that the channel level is unequal to 0.0% and that the corresponding From parameter (except for A/B) is greater than 0.0%.

I tested the arrow behavior recently and could not find any problem. But feel free to re-open the ticket if you think there is still something wrong.
